geo-rep: remove problematic language

itisravi opened the pull request:
Replace master and slave terminology in geo-replication with primary and
secondary respectively.

All instances are replaced in cli and glusterd.

Changes to other parts of the code to follow in separate patches.
tests/00-geo-rep/* are passing thus far.

Updates: #1415
Change-Id: Ifb12b7f5ce927a4a61bda1e953c1eb0fdfc8a7c5
Signed-off-by: Ravishankar N <ravishankar@redhat.com>

itisravi committed:
cli/glusterd: conscious language changes for geo-rep

Replace master and slave terminology in geo-replication with primary and
secondary respectively.

All instances are replaced in cli and glusterd.

Changes to other parts of the code to follow in separate patches.
tests/00-geo-rep/* are passing thus far.

Updates: #1415
Change-Id: Ifb12b7f5ce927a4a61bda1e953c1eb0fdfc8a7c5
Signed-off-by: Ravishankar N <ravishankar@redhat.com>

itisravi commented:
/run regression

itisravi commented:
/run regression

itisravi commented:
@deepshikhaaa The regression is getting aborted.

itisravi commented:
/run regression

deepshikhaaa commented:
/run regression

itisravi commented:
/run regression

itisravi commented:
/run regression

itisravi commented:
/run regression

aravindavk commented on geo-replication/gsyncd.conf.in:
```diff
@@ -220,17 +220,17 @@ template=true
 configurable=false
 
 [slave-log-file]
```
These things are not changed, planning to change in the next PR?

aravindavk replied:
If changed here then change also required in other places

* https://github.com/gluster/glusterfs/blob/devel/geo-replication/syncdaemon/gsyncdconfig.py#L234 and other places in this file.
* https://github.com/gluster/glusterfs/blob/devel/geo-replication/syncdaemon/resource.py

itisravi replied:
@aravindavk  yes, this patch only changes glusterd and cli completely, plus whatever is needed to make all the geo-rep .ts ( tests/bugs/geo-replication/* and /tests/00-geo-rep/* ) pass because of these glusterd/cli changes. I tested that the .ts pass on my local machine. I am sending further changes as separate patches on top of this patch.

aravindavk replied:
ack

aravindavk reviewed (dismissed):
Changes Looks good to me. Hope next PRs will cover some more occurrences of these terms.

Shwetha-Acharya reviewed (commented):
Please make sure to add the changes in hookscripts from  https://github.com/gluster/glusterfs/blob/devel/extras/hook-scripts/S56glusterd-geo-rep-create-post.sh#L82
Any specific reasons for missing this https://github.com/gluster/glusterfs/blob/devel/tests/00-geo-rep/01-georep-glusterd-tests.t#L179?

itisravi commented:
> Please make sure to add the changes in hookscripts from https://github.com/gluster/glusterfs/blob/devel/extras/hook-scripts/S56glusterd-geo-rep-create-post.sh#L82
> Any specific reasons for missing this https://github.com/gluster/glusterfs/blob/devel/tests/00-geo-rep/01-georep-glusterd-tests.t#L179?
@Shwetha-Acharya Please see my reply above, I will send additional changes on top of this patch.

kotreshhr commented on extras/hook-scripts/S56glusterd-geo-rep-create-post.sh:
```diff
@@ -1,8 +1,10 @@
-#!/bin/bash
+#!/ bin / bash
```
@itisravi this is the reason the geo-rep test case was timing out with your patch. Please fix and re-run.

itisravi replied:
@kotreshhr Thanks!

itisravi committed:
tests: remove offensive language

TODO:
Remove 'slave-timeout' and 'slave-gluster-command-dir'.
These variables are defined in geo-replication/gsyncd.conf.in.
So I will remove them when I change that folder.

Change-Id: Ib9167ca586d83e01f8ec755cdf58b3438184c9dd
Signed-off-by: Ravishankar N <ravishankar@redhat.com>

itisravi commented:
/run regression

itisravi commented:
/run regression

itisravi commented:
/recheck smoke

itisravi commented:
Just an update: Since I am making incremental commits with this PR, there are 2 patches so far:
 
```
2. tests: remove offensive language 
1. cli/glusterd: conscious language changes for geo-rep
```

With 1, all tests passed. With 2 on top of 1, everything passes except `00-geo-rep/georep-config-upgrade.t`  Working with @kotreshhr  to debug it (not successful so far). After fixing it, next patch on top of this is to change offensive occurrences in` geo-replication` folder and then `extras` folder.

itisravi committed:
geo-rep: remove offensive language

Change-Id: I9e34ccfd28029209262874993836dd36ad3c9c01
Signed-off-by: Ravishankar N <ravishankar@redhat.com>

itisravi committed:
extras, tools, doc: remove offensive language

Replace master and slave terminology in geo-replication with primary and
secondary respectively.

Change-Id: I3eb9242d2ce8340435265b764d28221d50f872c8
Signed-off-by: Ravishankar N <ravishankar@redhat.com>

itisravi commented:
/run regression

rakshithakamath94 commented:
/recheck smoke

itisravi commented:
/run regression

itisravi commented:
/run full regression

itisravi commented:
/run regression

itisravi commented on xlators/mgmt/glusterd/src/glusterd-store.c:
```diff
@@ -3014,6 +3014,29 @@ glusterd_store_retrieve_node_state(glusterd_volinfo_t *volinfo)
 
+/* change slave$i to secondary$i, where i = 1,2,3,...*/
+static void
+glusterd_chk_update_geo_rep_key_name(char **key)
```
Added this function to change the key slave1 to secondary1 in `/var/lib/glusterd/vols/$volname/info` during upgrade from old to new version, when glusterd is restarted.

itisravi commented:
/run regression

itisravi commented:
/recheck smoke

itisravi commented:
@Shwetha-Acharya @srijan-sivakumar Could help me test this patch to see if you encounter any issues during upgrade scenario? Sometimes when I try, I get these errors even without my patch, so I am not sure the are related to this PR:
```
gsyncd.log:[2020-12-22 08:06:36.199084] E [resource(worker /bricks/brick2):1312:service_loop] GLUSTER: Changelog History Crawl failed [{error=[Errno 0] Success}] 
changes-bricks-brick3.log: [2020-12-22 08:07:27.697982 +0000] E [MSGID: 132009] [gf-history-changelog.c:939:gf_history_changelog] 0-gfchangelog: wrong result [{for=end}, {start=1608624440}, {idx=163}]
```

These are the steps I tried on a single node setup:
1. On the latest devel branch, create a geo-rep session:
```
ldconfig /usr/local/lib
gluster v create vol1 replica 3 127.0.0.2:/bricks/brick{1..3} force
gluster v create vol2 replica 3 127.0.0.2:/bricks/brick{4..6} force
gluster v create gluster_shared_storage replica 3 127.0.0.2:/bricks/brick{7..9} force
gluster v start vol1
gluster v start vol2
gluster v start gluster_shared_storage
sleep 5
mkdir /var/run/gluster/shared_storage
mount -t glusterfs 127.0.0.2:vol1 /mnt/vol1
mount -t glusterfs 127.0.0.2:vol2 /mnt/vol2
mount -t glusterfs 127.0.0.2:gluster_shared_storage /var/run/gluster/shared_storage
sleep 5
gluster system:: exec gsec_create
sleep 5
gluster v geo-rep vol1 127.0.0.2::vol2 create push-pem
sleep 10
gluster vol geo-rep vol1 127.0.0.2::vol2 config use_meta_volume  true
sleep 10
gluster v geo-rep vol1 127.0.0.2::vol2 start
```
2.Create some files on `/mnt/vol1` and check that they are synced to `/mnt/vol2`.
3. Stop the session:`gluster v geo-rep vol1 127.0.0.2::vol2 stop`.
4. Create some files/folders on `/mnt/vol1`.
4. Kill all gluster processes, apply my patchet: `wget https://github.com/gluster/glusterfs/pull/1568.patch && git am 1568.patch`.
5. Compile, install and restart glusterd.
6. Fuse mount `/mnt/vol1`,`'/mnt/vol2` and  `/var/run/gluster/shared_storage` again.
7.  Restart the session: `gluster v geo-rep vol1 127.0.0.2::vol2 start`.
8. Ensure that `/mnt/vol2` has the newly created files/folders.

schaffung commented:
> @Shwetha-Acharya @srijan-sivakumar Could help me test this patch to see if you encounter any issues during upgrade scenario? Sometimes when I try, I get these errors even without my patch, so I am not sure the are related to this PR:
> 
> ```
> gsyncd.log:[2020-12-22 08:06:36.199084] E [resource(worker /bricks/brick2):1312:service_loop] GLUSTER: Changelog History Crawl failed [{error=[Errno 0] Success}] 
> changes-bricks-brick3.log: [2020-12-22 08:07:27.697982 +0000] E [MSGID: 132009] [gf-history-changelog.c:939:gf_history_changelog] 0-gfchangelog: wrong result [{for=end}, {start=1608624440}, {idx=163}]
> ```
> 
> These are the steps I tried on a single node setup:
> 
>     1. On the latest devel branch, create a geo-rep session:
> 
> 
> ```
> ldconfig /usr/local/lib
> gluster v create vol1 replica 3 127.0.0.2:/bricks/brick{1..3} force
> gluster v create vol2 replica 3 127.0.0.2:/bricks/brick{4..6} force
> gluster v create gluster_shared_storage replica 3 127.0.0.2:/bricks/brick{7..9} force
> gluster v start vol1
> gluster v start vol2
> gluster v start gluster_shared_storage
> sleep 5
> mkdir /var/run/gluster/shared_storage
> mount -t glusterfs 127.0.0.2:vol1 /mnt/vol1
> mount -t glusterfs 127.0.0.2:vol2 /mnt/vol2
> mount -t glusterfs 127.0.0.2:gluster_shared_storage /var/run/gluster/shared_storage
> sleep 5
> gluster system:: exec gsec_create
> sleep 5
> gluster v geo-rep vol1 127.0.0.2::vol2 create push-pem
> sleep 10
> gluster vol geo-rep vol1 127.0.0.2::vol2 config use_meta_volume  true
> sleep 10
> gluster v geo-rep vol1 127.0.0.2::vol2 start
> ```
> 
> 2.Create some files on `/mnt/vol1` and check that they are synced to `/mnt/vol2`.
> 3. Stop the session:`gluster v geo-rep vol1 127.0.0.2::vol2 stop`.
> 4. Create some files/folders on `/mnt/vol1`.
> 4. Kill all gluster processes, apply my patchet: `wget https://github.com/gluster/glusterfs/pull/1568.patch && git am 1568.patch`.
> 5. Compile, install and restart glusterd.
> 6. Fuse mount `/mnt/vol1`,`'/mnt/vol2` and `/var/run/gluster/shared_storage` again.
> 7. Restart the session: `gluster v geo-rep vol1 127.0.0.2::vol2 start`.
> 8. Ensure that `/mnt/vol2` has the newly created files/folders.

Sure Ravi..will try the steps out and let you know the observations..

Shwetha-Acharya commented:
> Changelog History Crawl failed

@itisravi On repeating above steps, I was unable to get any errors as stated above. For me the upgrade scenario was unerring.

itisravi commented:
> > Changelog History Crawl failed
> 
> @itisravi On repeating above steps, I was unable to get any errors as stated above. For me the upgrade scenario was unerring.

Thanks @Shwetha-Acharya . I guess it must be a setup issue on my VM.  @kotreshhr  If you are also okay with the changes, I guess we can merge the patch.

itisravi commented:
@kotreshhr @aravindavk Please review. Trying to close this PR in 2020 itself. :grin: 
If you are merging, please `Rebase and merge` instead of `Squash and merge` so that each of the commits appear individually on top of the previous one.

aravindavk commented:
> @kotreshhr @aravindavk Please review. Trying to close this PR in 2020 itself. 😁
> If you are merging, please `Rebase and merge` instead of `Squash and merge` so that each of the commits appear individually on top of the previous one.

ack. Will plan to complete the review by afternoon.

aravindavk reviewed (approved):
LGTM

kotreshhr commented:
> @kotreshhr @aravindavk Please review. Trying to close this PR in 2020 itself. :grin: 
> If you are merging, please `Rebase and merge` instead of `Squash and merge` so that each of the commits appear individually on top of the previous one.

I pretty much followed this PR. I will review and try to close it by tomorrow.

kotreshhr reviewed (approved):
Good work! LGTM.
